librz/include/rz_asm.h

```diff
@@ -198,6 +201,16 @@ RZ_API int rz_asm_op_set_hexbuf(RzAsmOp *op, const ut8 *buf, int len);
 RZ_API void rz_asm_op_set_buf(RzAsmOp *op, const ut8 *str, int len);
 RZ_API ut8 *rz_asm_op_get_buf(RzAsmOp *op);
 
+// String tokenizing
+RZ_API RZ_OWN RzAsmTokenString *rz_asm_token_string_new(const char *asm_str);
+RZ_API void rz_asm_token_string_free(RZ_OWN RzAsmTokenString *toks);
+RZ_API RZ_OWN RzAsmTokenString *rz_asm_token_string_clone(RZ_OWN RZ_NONNULL RzAsmTokenString *toks);
+RZ_API void rz_asm_token_pattern_free(void *p);
+RZ_API RZ_OWN RzAsmTokenString *rz_asm_tokenize_asm_regex(RZ_BORROW RzStrBuf *asm_str, RzPVector /* RzAsmTokenPattern* */ *patterns);
+RZ_API RZ_OWN RzAsmParseParam *rz_asm_get_parse_param(RZ_NULLABLE const RzReg *reg, ut32 ana_op_type);
+RZ_DEPRECATE RZ_API RZ_OWN RzAsmTokenString *rz_asm_tokenize_asm_string(RZ_BORROW RzStrBuf *asm_str, RZ_NULLABLE const RzAsmParseParam *param);
+RZ_DEPRECATE RZ_API RZ_OWN RzStrBuf *rz_asm_colorize_asm_str(RZ_BORROW RzStrBuf *asm_str, RZ_BORROW RzPrint *p, RZ_NULLABLE const RzAsmParseParam *param, RZ_NULLABLE const RzAsmTokenString *toks);
```

Comment:
The idea was to mark it as deprecated so people are encouraged to implement a custom token parser for a plugin. This way no there wouldn't be two ways of parsing around. @XVilka  brought it in if I recall it correctly. But since the generic parsing is rewritten completely we can change it back.

Reply:
Another advantage of nudging devs to the custom parsing method is to get rid of those dependencies between `RzAsm` <-> `RzReg` (like below). Because the plugin has all these information anyways.

Reply:
Yes, the idea is to discourage using generic parsing and move everything to plugins.